src/strat_engine/blockdev.rs

```diff
@@ -381,4 +381,33 @@ impl BlockDev {
     pub fn dstr(&self) -> String {
         format!("{}:{}", self.dev.major, self.dev.minor)
     }
+
+    /// List all used areas in the blockdev.
+    fn used_areas(&self) -> Vec<(SectorOffset, Sectors)> {
+        let mut used = Vec::new();
+
+        // Flag start mda zone as used
+        used.push((SectorOffset(0),
+                   BDA_STATIC_HDR_SIZE + self.mda_sectors + self.reserved_sectors));
+        // Flag end mda zone as used
+        used.push((SectorOffset(*self.sectors - *BDA_STATIC_HDR_SIZE - *self.mda_sectors),
+                   BDA_STATIC_HDR_SIZE + self.mda_sectors));
+        used
+    }
+
+    /// List all unused, available extents in this blockdev.
+    pub fn avail_areas(&self) -> Vec<(SectorOffset, Sectors)> {
+        let used = self.used_areas();
+        assert_eq!(used.len(), 2);
```

Comment:
Hmm. Well it really depends on where we keep track of allocations in the blockdev. In Froyo, the Blockdev tracked this itself, but maybe Stratis tracks this in the Flex layer, and so all Blockdev has to report is its reserved areas.

Ok, since [YAGNI](https://en.wikipedia.org/wiki/You_aren't_gonna_need_it) let me just have avail_area report the offset/length of its usable area (reduced by metadata areas) and we can wait and see how we end up tracking blockdev allocations.